File: Ruby/problem9.rb
#! /usr/bin/ruby

#A Pythagorean triplet is a set of three natural numbers, a < b < c, for which,
#
#a2 + b2 = c2
#For example, 32 + 42 = 9 + 16 = 25 = 52.
#
#There exists exactly one Pythagorean triplet for which a + b + c = 1000.
#Find the product abc.


timer_start = Time.now

1.upto(1000) do |c|
  1.upto(c) do |b|
    1.upto(b) do |a|
      if a * a + b * b == c * c
        if a + b + c == 1000
          puts "(#{a},#{b},#{c})" 
          answer = a * b * c
          puts answer
        end
      end
    end
  end
end


puts "Elapsed Time: #{(Time.now - timer_start)*1000} milliseconds"

File: JS/problem9.js
//A Pythagorean triplet is a set of three natural numbers, a < b < c, for which,
//a2 + b2 = c2
//For example, 32 + 42 = 9 + 16 = 25 = 52.

//There exists exactly one Pythagorean triplet for which a + b + c = 1000.
//Find the product abc.

var start = new Date().getTime();

var answer = 0;

for (var i = 1; i <= 1000; i ++){
  for (var j = 1; j<=1000; j++){
    for (var k = 1; k<= 1000; k++){
     if ((j * j) + (i * i) === (k * k) && i + j + k === 1000){
        console.log("i = " + i + " j = " + j + " k = " + k);
        answer = j * i * k
        console.log(answer);
     }
    }
  }
}

var end = new Date().getTime();
var time = end - start;
console.log('Execution time: ' + time);